src/WebJobs.Script/Rpc/LanguageWorkerChannel.cs

```diff
@@ -362,23 +365,33 @@ internal void InvokeResponse(InvocationResponse invokeResponse)
             }
         }
 
+        internal void SystemLog(RpcEvent msg)
+        {
+            var rpcLog = msg.Message.RpcSystemLog;
+            LogContents(rpcLog.InvocationId, $"SystemLog: {rpcLog.Message}", rpcLog.Exception.Message, rpcLog.Exception.StackTrace, null, (LogLevel)rpcLog.Level);
+        }
+
         internal void Log(RpcEvent msg)
         {
-            var rpcLog = msg.Message.RpcLog;
-            LogLevel logLevel = (LogLevel)rpcLog.Level;
-            if (_executingInvocations.TryGetValue(rpcLog.InvocationId, out ScriptInvocationContext context))
+            RpcLog rpcLog = msg.Message.RpcLog;
+            LogContents(rpcLog.InvocationId, rpcLog.Message, rpcLog.Exception.Message, rpcLog.Exception.StackTrace, rpcLog.EventId, (LogLevel)rpcLog.Level);
+        }
+
+        private void LogContents(string invocationId, string message, string exceptionMessage, string stackTrace, string eventId, LogLevel logLevel)
+        {
+            if (_executingInvocations.TryGetValue(invocationId, out ScriptInvocationContext context))
             {
                 // Restore the execution context from the original invocation. This allows AsyncLocal state to flow to loggers.
                 System.Threading.ExecutionContext.Run(context.AsyncExecutionContext, (s) =>
                 {
-                    if (rpcLog.Exception != null)
+                    if (exceptionMessage != null)
                     {
-                        var exception = new RpcException(rpcLog.Message, rpcLog.Exception.Message, rpcLog.Exception.StackTrace);
-                        context.Logger.Log(logLevel, new EventId(0, rpcLog.EventId), rpcLog.Message, exception, (state, exc) => state);
+                        var exception = new RpcException(message, exceptionMessage, stackTrace);
+                        context.Logger.Log(logLevel, new EventId(0, eventId), message, exception, (state, exc) => state);
```

Comment:
context.Logger.Log should be used only for invocation logs. Also, need to figure out if a language worker logs system log within the invocation context, is it possbile to populate invoation id, function name etc as part of the system log?